torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1781,6 +1781,14 @@ struct to_ir {
       Expr e = target.target();
 
       auto* rhs = emitExpr(e);
+      if (auto namedType = rhs->type()->cast<c10::NamedType>()) {
+        if (auto qualname = namedType->name()) {
+          if (*qualname ==
+              "__torch__.torch.fx.symbolic_trace.ModuleHierarchyCtxMgr") {
+            continue;
+          }
+        }
+      }
```

Comment:
@SplitInfinity any way I get get TorchScript to ignore `with` statements with my `ModuleHierarchyCtxMgr` object in a more elegant way?

Reply:
Nope :'(

Why are you trying to ignore them?

Reply:
Because they're non-functional except for in FX tracing, which is not supported on TS programs. It uses a global variable, so not TS compilable. I can try to split out the actual behavior into @ignore blocks